docs/proposals/policy.md

```diff
@@ -0,0 +1,438 @@
+
+## API
+### Basic CRUD
+The basic CRUD for Roles and RoleBindings, and Policy is not as straightforward as usual.  
+
+The REST API will respond to gets, lists, creates, updates, and deletes (not watches) at the expected locations for Roles and RoleBindings.  The REST API will respond to gets, lists, and watches (not creates, updates, and deletes) at the expected location for Policy.  
+
+Only one Policy document is allowed per namespace.  Roles and RoleBindings are not stored in etcd.  Instead, requests against their REST endpoints will result in inspection and modification of the Policy document.  This makes it easier to express restrictions on how policy can be manipulated, while making it possible to guarantee of referential integrity for references inside the local namespace.
+
+#### allowedentities?namespace=<namespace>
+**Please suggest a better name and a better location to put this** Given a namespace which users and groups can see or edit some resource contained in that namespace.  This API is not guaranteed to return accurate results.  It is possible that a user returned in the list, could be denied access to any resource in the namespace.  This can happen when a user is granted allow rights for something the namespace, but a deny rule later stomps on it.  Barring weird cases like that, it will work.  A user must have get permission on requested namespace's policy object in order to make this call.  The response will look like this:
+```
+{
+	"users": ["someuser"],
+	"groups": ["somegroup"]
+}
+```
+
+#### mothermayi
+**Please suggest a better name and a better location to put this.** Can the **authenticated** user perform a given request?  This API is guaranteed to be accurate.  Even though this is logically a GET the request will have to made as a POST in order to specify all the information in a reasonable way.  Authentication is the only requirement to call this API.  The request must specify an object like this:
+```
+{
+	"hypotheticalRequests": [
+		{
+			"httpVerb": "GET",
+			"path": "url/of/hypothetical/request",
+			"parameters": [
+				{"name": "param-name", "value": "param-value"}
+			],
+			"body": "body of hypothetical request"
+		}
+	]
+}
+```
+The The return mirrors the Authorizer.Authorize() return value:
+```
+{
+	"judgements": [
+		{
+			"allowedBy": "string reason for allowing",
+			"deniedBecause": "string reason for denying",
+			"error": "string describing why the server couldn't determine the answer"
+		}
+	]
+}
+```
+
+#### canuserdo
```

Comment:
This API is not needed if you just implement something like the ***SubjectAccessReview*** operation described in prior comment since that could take an optional **Subject**.  If you need to separate it for policy enforcement, I guess that is fine too, but then it should be structured like the prior comment, and this operation becomes **SubjectAccessReview**, and the earlier operation becomes **AuthenticatedUserAccessReview** or something similar.